src/controllers/connectionManager.ts

```diff
@@ -260,15 +260,18 @@ export default class ConnectionManager {
 	}
 
 	/**
-	 * Get the connection string for the provided connection Uri
-	 * @param connectionUri The connection Uri for the connection.
+	 * Get the connection string for the provided connection Uri or ConnectionDetails.
+	 * @param connectionUri The connection Uri for the connection. If undefined then a ConnectionDetails should be specified.
+	 * @param connectionDetails The connection details for the connection. If undefined then a connection URI should be specified.
 	 * @param includePassword (optional) if password should be included in connection string.
 	 * @param includeApplicationName (optional) if application name should be included in connection string.
 	 * @returns connection string for the connection
 	 */
-	public async getConnectionString(connectionUri: string, includePassword: boolean = false, includeApplicationName: boolean = true): Promise<string> {
+	public async getConnectionString(connectionUri?: string, connectionDetails?: ConnectionContracts.ConnectionDetails,
```

Comment:
It'd be better to make it `connectionUriOrDetails: string | ConnectionContracts.ConnectionDetails` - less chance of someone calling this incorrectly by not passing in either a URI or details